gitlab/utils.py

```diff
@@ -90,3 +93,30 @@ def __new__(  # type: ignore
 
 def remove_none_from_dict(data: Dict[str, Any]) -> Dict[str, Any]:
     return {k: v for k, v in data.items() if v is not None}
+
+
+def warn(
+    *,
+    message: str,
+    category: Optional[Type] = None,
+    source: Optional[Any] = None,
+) -> None:
+    """This `warnings.warn` wrapper function attempts to show the location causing the
+    warning in the user code that called the library.
+
+    It does this by walking up the stack trace to find the first frame located outside
+    the `gitlab/` directory. This is helpful to users as it shows them their code that
+    is causing the warning.
+    """
+    # Get `stacklevel` for user code so we indicate where issue is in
+    # their code.
+    pg_dir = os.path.abspath(os.path.dirname(__file__))
```

Comment:
Apart from some legacy config files I believe we've mostly already migrated everything to `pathlib`. 
This would be something like `Path(__file__).parent.resolve()` I think. Then you can use `is_relative_to()` later on.

```suggestion
    gitlab_dir = Path(__file__).parent.resolve()
```

Reply:
Looks nice. Will do.

Reply:
Done

Reply:
> Apart from some legacy config files I believe we've mostly already migrated everything to `pathlib`. This would be something like `Path(__file__).parent.resolve()` I think. Then you can use `is_relative_to()` later on.

Darn. No `is_relative_to()` until Python 3.9. 😟